### PR TITLE
Fix multiplane sampler conversion test

### PR DESCRIPTION
**Pull request by ziga-lunarg**
Test failed on AMD as described in #3362

### CODE REVIEW

File: tests/vklayertests_buffer_image_memory_sampler.cpp [hunk 1]
```diff
@@ -10307,6 +10307,11 @@ TEST_F(VkLayerTest, MultiplaneImageSamplerConversionMismatch) {
                                            {0, VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER, 2, VK_SHADER_STAGE_ALL, samplers},
                                        });
 
+    if (!descriptor_set.set_) {
```
Comment (ncesario-lunarg) [inline tests/vklayertests_buffer_image_memory_sampler.cpp:10310]: I would feel a little better if https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/a3c47d193b33fb309876dad239bc9985c6f7d2c7/tests/layer_validation_tests.h#L409 explicitly set `set_ = VK_NULL_HANDLE`. Though, maybe `set_{}` accomplishes this at https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/a3c47d193b33fb309876dad239bc9985c6f7d2c7/tests/layer_validation_tests.cpp#L1506.

**Review by ncesario-lunarg — APPROVED** [review]
One question, but LGTM!

**Review by Tony-LunarG — APPROVED** [review]
LGTM - thanks!

File: tests/vklayertests_buffer_image_memory_sampler.cpp (hunk 1 above)
Comment (ziga-lunarg) [reply tests/vklayertests_buffer_image_memory_sampler.cpp:10310]: I have updated the constructor to explicitly initialize `VkDescriptorSet` as `VK_NULL_HANDLE`, but at least on the driver that this test fails on `vk::AllocateDescriptorSets()` overwrites it to `VK_NULL_HANDLE` anyway when it fails.

**Pull request merged**